aten/src/ATen/native/quantized/affine_quantizer_base.cpp

```diff
@@ -118,10 +118,20 @@ uint8_t quantize_val_arm(
     const float scale,
     const int32_t zero_point,
     const float value) {
-  const int32_t qmin = std::numeric_limits<uint8_t>::min();
-  const int32_t qmax = std::numeric_limits<uint8_t>::max();
+  constexpr int32_t qmin = std::numeric_limits<uint8_t>::min();
+  constexpr int32_t qmax = std::numeric_limits<uint8_t>::max();
   float inv_scale = 1.0f / scale;
+#ifndef _MSC_VER
+  auto r = static_cast<int32_t>(Round(value * inv_scale));
+  // buiiltin_add_overflow() returns true in case of overflow
+  if (__builtin_add_overflow(zero_point, r, &r)) {
+    // zero_point must be a non-negative value between qmin and qmax,
+    // i.e. only overflow can happen.
+    r = qmax;
+  }
+#else
   auto r = zero_point + static_cast<int32_t>(Round(value * inv_scale));
```

Comment:
Same can be fixed by moving `zero_point +` before value is rounded, but it might have some negative effects on accruacy.

Reply:
Is this a test failure that is being fixed? Because very likely the result of Round + zero_point will fit within int32_t `value * inv_scale` will likely be not a large value. That said mathematically what you did sound right but we can get the same by `auto r = static_cast<int64_t>(zero_point) + static_cast<int64_t>(Round(value * inv_scale));`, right? and in 64bit system this would be ok perf wise which would be safe to do even when _MSC_VER.

Reply:
Do you know why all `quantized_val` functions follow the same pattern of rounding first and then adding zero_point?

Reply:
For affine quantization that is the case: https://github.com/google/gemmlowp/blob/master/doc/quantization.md. Not sure if that is what you are asking.

Reply:
Ok, updated the logic from `auto r = zero_point + static_cast<int32_t>(Round(value * inv_scale));` to `auto r = static_cast<int32_t>(zero_point + Round(value * inv_scale));`